test/port-finder-0-vs-127-test.js

```diff
@@ -0,0 +1,47 @@
+/*
+ * port-finder-0-vs-127-test.js: Test for the `portfinder` module.
+ * that demonstrates issue #24 
+ * https://github.com/indexzero/node-portfinder/issues/24
+ */
+
+var vows = require('vows'),
+    assert = require('assert'),
+    portfinder = require('../lib/portfinder'),
+    child_process = require('child_process'),
+    path = require('path'),
+    http = require('http');
+
+const host = "localhost";
+var server;
+
+vows.describe('portfinder').addBatch({
+  "When using portfinder module": {
+    "ensure that infinite loop is not triggered": {
+      topic: function () {
+    var callback = this.callback;
+    portfinder.getPort(function(err, port) {
+      server = http.createServer(function(){}).listen(port, host, function() {
+        var timeout = false;
+        var fileToExec = path.join(__dirname, 'getPort.js');
+        var timer = setTimeout(function() {
+          timeout = true;
+          process.kill(child.pid);
+          callback(null, "timeout");  
+        }, 10000); // 10 seconds
+        var child = child_process.spawn('node', [fileToExec,  port]);
+        child.on('close', function() {
+          if (timeout === false) {
+            clearTimeout(timer);
+            callback(null);
+          }
+        });
+      });  
+    });
+      },
+      "should not get a timeout": function (err, res) {
+        server.close();
+        assert.isFalse(!!res);
+      }
+    }
+  }
+}).export(module);
```

Comment:
so a couple of things - most are my fault, this is pretty creative may have its merits - that said...

a) if we do keep it, can we rename the file to port-finder-integration-test.js
b) im not sure we need it - we have a similar setup already - we can add localhost to this array: https://github.com/indexzero/node-portfinder/blob/master/test/helper.js#L27 - i'd actually like to add an exact duplicate ip address there (you can add localhost as well, it should not matter, if our expected behavior works out right). This will test that we can pass any hostname, even if mapped to the same ip, more than once to our input, preventing pretty much all the bugs that have risen since 1.0

I'd also like to catch the error that only travis throws in that file, which is why I added these lines: https://github.com/indexzero/node-portfinder/blob/master/test/helper.js#L14-L16 - in this case, travis is just another user, he/she just happens to be setup differently - we should handle that error in our iterator(if it becomes hard, we'll file an issue for that, dont worry to much)

let me know your thoughts on all of this, thanks again for your help - if you want to hand over the work, i understand - that said, if you want to bang it out, go for it, you've done good work